src/qtism/data/storage/xml/Utils.php

```diff
@@ -358,15 +362,26 @@ public static function valueAsString($value, $encode = true): string
         return (string)$value;
     }
 
+    public static function escapeUnicodeForXML($string): string
+    {
+        return preg_replace_callback(
+            '/[^\x09\x0A\x0D\x20-\x7F]/',
+            function ($matches) {
+                return sprintf('&#x%04X;', ord($matches[0]));
```

Comment:
This will produce an HTML entity. XML doesn't support these.
<img width="439" alt="image" src="https://github.com/user-attachments/assets/823e6a0f-1f7e-4ad1-b2dd-7bd65a563752">

Reply:
You can see that the changes from this PR do not work as intended [here](https://oat-sa.atlassian.net/browse/TR-6347?focusedCommentId=285790).